src/hdmf/container.py

```diff
@@ -735,14 +738,29 @@ def _generate_field_html(self, key, value, level, access_code):
 
 
     def _generate_array_html(self, array, level):
-        """Generates HTML for array data"""
+        """Generates HTML for array data (e.g., NumPy arrays, HDF5 datasets, Zarr datasets and DataIO objects)."""
 
-        read_io = self.get_read_io()  # if the Container was read from file, get IO object
-        if read_io is not None: # Note that sometimes numpy array have a read_io attribute
-            repr_html = read_io.generate_dataset_html(array)
-        else:
+        is_numpy_array = isinstance(array, np.ndarray)
+        it_was_read_with_io = self.get_read_io() is not None
```

Comment:
```suggestion
        read_io = self.get_read_io()
        it_was_read_with_io = read_io is not None
```

This is just to avoid having to call `self.get_read_io` twice. The `io` object is actually only stored on the root container (i.e., `NWBFile`). As such `get_read_io` has to iterate through the parent hierarchy to find the `io` object. This probably just minor, but I think it could help in some cases with performance to avoid repeating this search.